pkg/providers/vsphere/virtualmachine/cdrom.go

```diff
@@ -235,6 +214,59 @@ func getBackingFileNameByImageRef(
 	return itemStatus.FileInfo[0].StorageURI, nil
 }
 
+// processImage validates if the image is ready and of type ISO, and returns the
+// content library item UUID and status from its provider ref.
+func processImage(
+	ctx context.Context,
+	client ctrlclient.Client,
+	imageName string,
+	namespace string) (string, imgregv1a1.ContentLibraryItemStatus, error) {
+
+	if namespace != "" {
```

Comment:
Looks like `VirtualMachineImage` and `ClusterVirtualMachineImage` have the same fields, just a different name. I think you can reduce some the duplication like so:

```go
	var vmi vmopv1.VirtualMachineImage

	if namespace == "" {
		// Cluster scope image.
		var cvmi vmopv1.ClusterVirtualMachineImage
		if err := client.Get(ctx, ctrlclient.ObjectKey{Name: imageName, Namespace: namespace}, &cvmi); err != nil {
			return "", imgregv1a1.ContentLibraryItemStatus{}, err
		}
		vmi = vmopv1.VirtualMachineImage(cvmi)
	} else {
		// Namespace scope image.
		if err := client.Get(ctx, ctrlclient.ObjectKey{Name: imageName}, &vmi); err != nil {
			return "", imgregv1a1.ContentLibraryItemStatus{}, err
		}
	}

	if !conditions.IsTrue(&vmi, vmopv1.ReadyConditionType) {
		return "", imgregv1a1.ContentLibraryItemStatus{}, fmt.Errorf("image condition is not ready: %v", conditions.Get(&vmi, vmopv1.ReadyConditionType))
	}
	if vmi.Status.Type != string(imgregv1a1.ContentLibraryItemTypeIso) {
		return "", imgregv1a1.ContentLibraryItemStatus{}, fmt.Errorf("image type %q is not ISO", vmi.Status.Type)
	}
	if vmi.Spec.ProviderRef == nil {
		return "", imgregv1a1.ContentLibraryItemStatus{}, errors.New("image provider ref is nil")
	}
	var cclitem imgregv1a1.ClusterContentLibraryItem
	if err := client.Get(ctx, ctrlclient.ObjectKey{Name: vmi.Spec.ProviderRef.Name}, &cclitem); err != nil {
		return "", imgregv1a1.ContentLibraryItemStatus{}, err
	}

	return string(cclitem.Spec.UUID), cclitem.Status, nil
```

Reply:
Thanks for the suggestion! The cl-item could also be namespace or cluster scope, so I applied the same logic to retrieve it. PTAL https://github.com/vmware-tanzu/vm-operator/compare/f963ecdd2927200a128547ccea69bcec4035dc3f..86af91903d74df668854745b36a7a06b2c43ae22

Reply:
Nice!